Source/inv.cpp

```diff
@@ -2189,7 +2162,7 @@ void AutoGetItem(int pnum, int ii)
 	}
 
 	if (done) {
-		CleanupItems(ii);
+		CleanupItems(&plr[pnum].HoldItem, ii);

```

Comment:
this would previously have been items[ii], so could you explain this change to me ?

Reply:
They are the same. A little bit above we assign holditem to items[ii]

Reply:
Oh i see now on line 2143, I hade missed that previously when going over despite looking.

Reply:
Admittedly, this is not very intuitive... it all goes back to the damn "holditem". It should be used to represent "the item in the cursor" but it's used as this temporary storage everywhere. Once we get rid of it and make those "lists of ints" lists of pointers, i think this kind of stuff will be dramatically simplified.

One step at a time though...